x/claim/keeper/invariants.go

```diff
@@ -44,3 +53,21 @@ func AirdropSupplyInvariant(k Keeper) sdk.Invariant {
 		return "", false
 	}
 }
+
+// ClaimRecordInvariant invariant checks that claim record was claimed but not completed
+func ClaimRecordInvariant(k Keeper) sdk.Invariant {
+	return func(ctx sdk.Context) (string, bool) {
+		missions := k.GetAllMission(ctx)
+		claimRecords := k.GetAllClaimRecord(ctx)
+
+		for _, claimRecord := range claimRecords {
+			for _, mission := range missions {
+				if !claimRecord.IsMissionCompleted(mission.MissionID) &&
+					claimRecord.IsMissionClaimed(mission.MissionID) {
+					return fmt.Sprintf("mission %d claimed but not completed", mission.MissionID), true
+				}
+			}
```

Comment:
What if one the mission ID listed in the `claimRecord` doesn't exists in `missions`? 

In other words, should we consider database inconsistency as an *invariant* ?

Reply:
yes!! I was thinking the same. An invariant can be good in this case

Reply:
https://github.com/ignite/modules/issues/80